### PR TITLE
Support inline error tooltip at cursor position

### PR DESCRIPTION
**Pull request by zhengbli**
The added behavior is quite similar to what we had before. But when the sublime version supports Phantom APIs (> 3118), we show a phantom popup instead of put the error message at the status bar (**only after you click**). The popup will go away when either the error is fixed or when the cursor (**the keyboard cursor** not the mouse cursor) moved to somewhere else. While the "on_hover" tooltip will always show the quick info.

In summary, the workflow is:

* hover with your mouse cursor to get quick info
* click at an error position to see error messages
* errors messages will appear when keyboard cursor is on top of an error span
* click elsewhere to dismiss an error message

gif

### CODE REVIEW

File: typescript/commands/error_info.py [hunk 1]
```diff
@@ -21,7 +21,11 @@ def run(self, text):
...
                 error_text = text
                 break
+
         if len(error_text) > 0:
-            self.view.set_status("typescript_error", error_text)
-        else:
-            self.view.erase_status("typescript_error")
\ No newline at end of file
+            if PHANTOM_SUPPORT:
+                template = '<body><style>div.error {{ background-color: brown; padding: 5px; }}</style><div class="error">{0}</div></body>'
```
Comment (billti) [inline typescript/commands/error_info.py]: Does the coloring change at all with the themes? Seeing as you only set background here, and not foreground, could foreground ever default to something in some themes which is unreadable?

File: typescript/commands/error_info.py [hunk 2]
```diff
@@ -21,7 +21,11 @@ def run(self, text):
...
+                display_text = template.format(error_text)
+                self.view.add_phantom("typescript_error", self.view.sel()[0], display_text, sublime.LAYOUT_BLOCK)
```
Comment (billti) [inline typescript/commands/error_info.py]: Interesting I don't see `add_phantom` on the `view` class in the docs (https://www.sublimetext.com/docs/3/api_reference.html#sublime.View). Maybe an oversight. They didn't make changes to this API after introducing it right?

File: typescript/commands/error_info.py [hunk 3]
```diff
@@ -21,7 +21,11 @@ def run(self, text):
             if region.contains(pt):
                 error_text = text
                 break
+
         if len(error_text) > 0:
-            self.view.set_status("typescript_error", error_text)
-        else:
-            self.view.erase_status("typescript_error")
\ No newline at end of file
+            if PHANTOM_SUPPORT:
```
Comment (billti) [inline typescript/commands/error_info.py]: If `show_errors_inline` is set to `false`, do phantoms still appear? As this is an `if/else`, that would mean no error info. Is there a downside to always updating the status bar anyway?

Comment (zhengbli) [inline typescript/commands/error_info.py]: Yes, `show_errors_inline` only affects sublime's own error tips, it doesn't toggle all phantom tooltips. Though I agree we can always update the status bar anyway.

**Comment by billti** [comment]
👍

**Comment by camloken** [comment]
It would be cool if you could place links in the tooltip that take you to the website rule. Atom does this.
